### ISSUE TITLE
Generates Filenames with special chars

### ISSUE THREAD

**Issue by Apollon77** [typedoc2md/typedoc-plugin-markdown#611]
### What package is the bug related to?

typedoc-plugin-markdown

### Describe the issue

I used the 4.0.1 on https://github.com/project-chip/matter.js with the updated config in https://github.com/project-chip/matter.js/pull/875 and got this warning:

> Some of the file names contain characters that are not allowed on Windows: 'TlvObject"'. If you commit, it might not be possible to check out the Git repository on Windows.

So it seems he used double quotes in a filename ... in fact I locally then found various files in a directory called " in the namespaces directory

find . -name '"*'
./docs/matter-node-ble.js/-internal-/namespaces/"
./docs/matter-node.js/exports/cluster/-internal-/namespaces/"
./docs/matter-node.js/exports/certificate/-internal-/namespaces/"
./docs/matter-node.js/exports/datatype/-internal-/namespaces/"
./docs/matter-node.js/exports/schema/-internal-/namespaces/"

After that also lookin in the error sof process if found eg

[warning] Failed to resolve link to "MatterSpecification.v10.Core" in comment for <internal>."/Users/ingof/Dev/GitHub/official-matter.js/packages/matter.js/dist/esm/tlv/TlvObject".ObjectSchema.

Something is wrong here and he created a "full path" directory somehow.

The repo told above is a public one, so you can easiely test yourself
 

### TypeDoc configuration

see https://github.com/project-chip/matter.js/pull/875 

basically:

"typedocOptions": {
        "entryPoints": [...],
        "plugin": ["typedoc-plugin-missing-exports", "typedoc-plugin-markdown"],
        "excludeExternals": true,
        "parametersFormat": "table",
        "membersWithOwnFile": ["Class", "Enum", "Interface"],
        "out": "..."
    }

### Expected behavior

No invalid paths/filenames are generated

**Comment by tgreyuk** [comment]
Thanks. 

So its included a fullpath in the name because the symbol represents a module that's been imported from the local file system. This is actually the typescript symbol name created when converting the project. Now there is a bug here (which will be fixed) but the fix will be to normalize the name so it doesn't try to create a weird file path. 

If you don't want the fullpath in the module name what you can do is to override the module name as follows:

packages/matter.js/src/tlv/TlvObject.ts

![Screenshot 2024-05-10 at 14 54 57](https://github.com/typedoc2md/typedoc-plugin-markdown/assets/11680870/9622f647-6bb4-4b99-8197-b338c3994849)

**Comment by Apollon77** [comment]
Hm ... Why this was no issue before 4.0? How to achieve the same result as in 3.x?

**Comment by tgreyuk** [comment]
> Hm ... Why this was no issue before 4.0? How to achieve the same result as in 3.x?

The module name is not a new issue (see output from v3). This is the resolved name from the ts compiler and also what you will see in the default html theme.

![Screenshot 2024-05-10 at 16 50 45](https://github.com/typedoc2md/typedoc-plugin-markdown/assets/11680870/61d05825-cad7-4e58-9742-cd0f17135162)

As I say - there will be a fix to the file paths coming shortly.

**Comment by tgreyuk** [comment]
Fix to file paths is available in typedoc-plugin-markdown@4.0.2. As previouslly mentioned the file path in the symbol name is not coming from this plugin.

**Comment by Apollon77** [comment]
Ok, thank you!
